Tasks/JavaToolInstallerV0/javatoolinstaller.ts

```diff
@@ -62,7 +62,8 @@ async function getJava(versionSpec: string, jdkArchitectureOption: string): Prom
                 taskLib.getInput('azureStorageAccountName', true), taskLib.getInput('azureContainerName', true), "");
             await azureDownloader.downloadArtifacts(extractLocation, '*' + fileNameAndPath);
             await taskutils.sleepFor(250); //Wait for the file to be released before extracting it.
-            jdkFileName = path.join(extractLocation, fileNameAndPath);
+            let jdkArchiveName = fileNameAndPath.slice(fileNameAndPath.lastIndexOf('/') + 1);
```

Comment:
Does it fix the issue generally? Could the azureCommonVirtualFile contain several nested folders?
Please use path module while operating with paths.